### PR TITLE
keeps sender and target both in fallback userop

### PR DESCRIPTION
**Pull request by livingrockrises**

### CODE REVIEW

File: contracts/gas-tank/SingletonGasTank.sol [hunk 1]
```diff
@@ -130,21 +131,25 @@ contract SingletonGasTank is Ownable, ReentrancyGuard {
...
          uint256 gasStarted = gasleft();
+         address target;
          require(msg.sender == tx.origin,"only EOA relayer");
          address payable relayer = payable(msg.sender);
          
         _validateSignature(fallbackUserOp);
         _validateAndUpdateNonce(fallbackUserOp);
 
-        (success, ret) = fallbackUserOp.sender.call{gas : fallbackUserOp.callGasLimit}(fallbackUserOp.callData);
+        if(fallbackUserOp.sender == fallbackUserOp.target) {
```
Comment (tomarsachin2271) [inline contracts/gas-tank/SingletonGasTank.sol:146]: We can take the target as fallbackUserOp.target always... no need for this if check.. and in both cases value sent in fallbackUserOp.taget field is used anyways.

File: contracts/gas-tank/SingletonGasTank.sol [hunk 2]
```diff
@@ -107,7 +107,8 @@ contract SingletonGasTank is Ownable, ReentrancyGuard {
     public pure returns (bytes32) {
         //can't use userOp.hash(), since it contains also the paymasterAndData itself.
         return keccak256(abi.encode(
-                fallbackUserOp.getSender(),
+                fallbackUserOp.sender,
+                fallbackUserOp.target,
                 fallbackUserOp.nonce,
```
Comment (tomarsachin2271) [inline contracts/gas-tank/SingletonGasTank.sol:112]: Did we compare gas fee comparison of taking nonce from storage while creating the hash and then just incrementing the nonce Vs taking nonce from userOp and then comparing the nonce later and then incrementing it?

Comment (livingrockrises) [reply contracts/gas-tank/SingletonGasTank.sol:112]: when you use getSender()

SingletonGasTank  ·  handleFallbackUserOp        ·     111519  ·     387959  ·     251473

when you access .sender

SingletonGasTank  ·  handleFallbackUserOp        ·     111672  ·     388112  ·     251626

later is about ~150 more gas units which I think is not significant.

File: contracts/gas-tank/SingletonGasTank.sol (hunk 1 above)
Comment (livingrockrises) [reply contracts/gas-tank/SingletonGasTank.sol:146]: ok

Comment (livingrockrises) [reply contracts/gas-tank/SingletonGasTank.sol:146]: done

**Pull request merged**